cmd/dlv/cmds/commands.go

```diff
@@ -409,9 +410,7 @@ func dapCmd(cmd *cobra.Command, args []string) {
 		}
 		defer logflags.Close()
 
-		if headless {
-			fmt.Fprintf(os.Stderr, "Warning: dap mode is always headless\n")
```

Comment:
I think we should use the `FlagSet.Changed` method to see if the user tried to explicitly set this to `false` and print a warning that it will be ignored.

Reply:
Just what I was looking for! Done!